c2corg_ui/__init__.py

```diff
@@ -68,3 +73,19 @@ def _add_static_view(config, name, path):
     )
     config.add_cache_buster(path, version_cache_buster)
     CACHE_PATH.append(name)
+
+
+class NotFound():
+
+    notFound = 'c2corg_ui:templates/404.html'
+
+    def __init__(self, request):
+        self.settings = request.registry.settings
+
+    @view_config(context=HTTPInternalServerError, renderer=notFound)
+    @view_config(context=HTTPBadRequest, renderer=notFound)
+    @notfound_view_config(renderer=notFound)
+    def notfound(self):
+        return {
```

Comment:
Untested, but in the [docs](http://docs.pylonsproject.org/projects/pyramid/en/latest/narr/viewconfig.html#view-config-placement) it sounds like you can get the `context`:

> When the decorator is used against a method of a class, a view is registered for the class, so the class constructor must accept an argument list in one of two forms: either a single argument, request, or two arguments, context, request.

Something like:

```
    def __init__(self, context, request):
        self.context = context
        self.settings = request.registry.settings
```